remotes/docker/config/hosts.go

```diff
@@ -232,13 +239,18 @@ func ConfigureHosts(ctx context.Context, options HostOptions) docker.RegistryHos
 				rhosts[i].Authorizer = authorizer
 			}
 
-			// When TLS has been explicitly configured for the operation or host, use the
-			// docker.HTTPFallback roundtripper to catch TLS errors and re-attempt the request as http.
-			// This allows preference for https when configured but also catches TLS errors early enough
-			// in the request to avoid sending the request twice or consuming the request body.
+			// When TLS has been explicitly configured for the operation or host and
+			// the protocol from the port number is ambiguous, use the
+			// docker.HTTPFallback roundtripper to catch TLS errors and re-attempt the
+			// request as http. This allows preference for https when configured but
+			// also catches TLS errors early enough in the request to avoid sending
```

Comment:
will this "flavor" pass through to enhance the error messaging and let the error show as trying https first based on (tryTLSFirst) is enabled?

would be nice if the error messaging was obvious telling the user what to do to address config changes.. for each testable situation

Reply:
Added an info log for this case